src/v2/BaseLoginRouter.js

```diff
@@ -215,19 +210,33 @@ export default Router.extend({
         .then(idxResp => {
           this.settings.unset('stateToken');
           this.settings.unset('proxyIdxResponse');
+          // TODO: do not unset  `useInteractionCodeFlow` OKTA-405474
           this.settings.unset('useInteractionCodeFlow');
           this.appState.trigger('remediationSuccess', idxResp);
-          this.render(Controller, options);
         })
         .catch(errorResp => {
           this.settings.unset('stateToken');
           this.settings.unset('proxyIdxResponse');
+          // TODO: do not unset  `useInteractionCodeFlow` OKTA-405474
           this.settings.unset('useInteractionCodeFlow');
           this.appState.trigger('remediationError', errorResp.error || errorResp);
-          this.render(Controller, options);
+        })
+        .finally(() => {
+          this._render(Controller, options);
         });
     }
 
+    this._render(Controller, options);
+  },
+
+  _render: function(Controller, options = {}) {
+    // Since we have a wrapper view, render our wrapper and use its content
```

Comment:
can you explain why we need this in comments, instead what it is doing?

Reply:
I just moved these lines from `render` method into `_render`

Reply:
i mean why it needs to be in `render`